src/components/MyAccount/NewSettings/EmailForm.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import EmailForm from "./EmailForm"
+import { filteredPickupLocations } from "../../../../__test__/fixtures/processedMyAccountData"
+import { PatronDataProvider } from "../../../context/PatronDataContext"
+import { processedPatron } from "../../../../__test__/fixtures/processedMyAccountData"
+
+describe("email form", () => {
+  const mockSetIsLoading = jest.fn()
+  const mockSetIsSuccess = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    global.fetch = jest.fn().mockResolvedValue({
+      json: async () => {
+        console.log("Updated")
+      },
+      status: 200,
+    } as Response)
+  })
+
+  const component = (
+    <PatronDataProvider
+      value={{
+        patron: processedPatron,
+        pickupLocations: filteredPickupLocations,
+      }}
+    >
+      <EmailForm
+        patronData={processedPatron}
+        setIsLoading={mockSetIsLoading}
+        setIsSuccess={mockSetIsSuccess}
+      />
+    </PatronDataProvider>
+  )
+
+  it("renders correctly with initial email", () => {
+    render(component)
+
+    expect(screen.getByText("streganonna@gmail.com")).toBeInTheDocument()
+
+    expect(screen.getByRole("button", { name: /edit/i })).toBeInTheDocument()
+  })
+
+  it("allows editing when edit button is clicked", () => {
+    render(component)
+    fireEvent.click(screen.getByRole("button", { name: /edit/i }))
+
+    expect(screen.getAllByLabelText("Update email")[0]).toBeInTheDocument()
+    expect(
+      screen.getByDisplayValue("streganonna@gmail.com")
+    ).toBeInTheDocument()
+
+    expect(screen.getByRole("button", { name: /cancel/i })).toBeInTheDocument()
+    expect(
+      screen.getByRole("button", { name: /save changes/i })
+    ).toBeInTheDocument()
+  })
+
+  it("validates email input correctly", () => {
+    render(component)
+
+    fireEvent.click(screen.getByRole("button", { name: /edit/i }))
+
+    const input = screen.getAllByLabelText("Update email")[0]
+    fireEvent.change(input, { target: { value: "invalid-email" } })
+
+    expect(
+      screen.getByText("Please enter a valid email address.")
+    ).toBeInTheDocument()
+
+    expect(screen.getByRole("button", { name: /save changes/i })).toBeDisabled()
+  })
+
+  it("allows adding a new email field", () => {
+    render(component)
+
+    fireEvent.click(screen.getByRole("button", { name: /edit/i }))
+
+    fireEvent.click(
+      screen.getByRole("button", { name: /\+ add an email address/i })
+    )
+
+    expect(screen.getAllByLabelText("Update email").length).toBe(3)
```

Comment:
this could be made less brittle if you expect it to be `patron.emails.length + 1`